### ISSUE TITLE
'acos' missing in PPF feature calculation

### ISSUE THREAD

**Issue by alfonsoros88** [PointCloudLibrary/pcl#1171]
Hi everyone,

in the ppf.cpp file, in the function computePPFPairFeature the 'acos' should be applied to the f1, f2 and f3 components of the feature vector. In the paper, these components should be angles between 0 and pi, but in the function they are leaved as the inner products of the vectors.

The problem arises when using the PPFRegistration class, these features are discretized as if they were angles. This could lead to unexpected behavior.

Best,
Alfonso

**Comment by taketwo** [comment]
But this function is not in use, isn't it?

Ping to the author @aichim, perhaps he can comment.

**Comment by alfonsoros88** [comment]
That's right! and that's an other issue I opened some weeks ago. The PPFEstimation class is not estimating ppfs features but pfh features.

**Comment by taketwo** [comment]
#1131

Well, I guess author of the code/changes should comment.

**Comment by Lanns** [comment]
yeah,i found the same problem too.
why dont you make some change for it?

**Comment by waschbaer00** [comment]
I also want to complain on this...

**Comment by JoyMazumder** [comment]
Can anyone fix this problem ? I am new in PCL and C++. So, I am facing some problem to solve it. Can any one share a working code ?

**Comment by taketwo** [comment]
No, the problem is still there. In fact, there are multiple other issues with PPF-based registration in PCL, so I would not recommend to use it.

**Comment by fersolerv** [comment]
Ok, thank you for the answer.
